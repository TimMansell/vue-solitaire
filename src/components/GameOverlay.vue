<template>
  <div class="winner" data-test="game-overlay">
    <div class="winner__content">
      <h1 class="winner__title">Sorry, no more Moves!</h1>
      <div class="winner__btn" @click="setGameLost(false)" data-test="game-overlay-btn">
        <NewGame />
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import NewGame from '@/components/NewGame.vue';

export default {
  name: 'GameOverlay',
  components: {
    NewGame,
  },
  methods: {
    ...mapActions(['setGameLost']),
  },
};
</script>

<style lang="scss" scoped>
.winner {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.7);
  z-index: 1;

  &__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 2rem;
  }

  &__title {
    color: white;
    text-shadow: -1px -1px rgba(0, 0, 0, 0.3);
    text-align: center;
  }
}
</style>
