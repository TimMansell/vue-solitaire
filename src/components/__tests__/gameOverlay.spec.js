import { shallowMount } from '@vue/test-utils';
import GameOverlay from '@/components/GameOverlay.vue';

describe('Card.vue', () => {
  it('matches visible snapshot', () => {
    const wrapper = shallowMount(GameOverlay);

    expect(wrapper).toMatchSnapshot();
  });
});
