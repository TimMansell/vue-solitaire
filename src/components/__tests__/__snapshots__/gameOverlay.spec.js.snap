// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue matches visible snapshot 1`] = `
<div data-test="game-overlay" class="game-overlay">
  <div class="game-overlay__content">
    <h1 class="game-overlay__title"></h1>
    <div data-test="game-overlay-btn" class="game-overlay__btn">
      <newgame-stub></newgame-stub>
    </div>
  </div>
</div>
`;
